src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/rollover/AttemptRolloverStep.kt

```diff
@@ -0,0 +1,152 @@
+            return
+        }
+
+        val alias = getAlias()
+
+        alias ?: return
+
+        val statsResponse = getIndexStats()
+
+        statsResponse ?: return
+
+        val indexCreationDate = Instant.ofEpochMilli(clusterService.state().metaData().index(managedIndexMetaData.index).creationDate)
+        val numDocs = statsResponse.primaries.docs.count
+        val indexSize = ByteSizeValue(statsResponse.primaries.docs.totalSizeInBytes)
+
+        if (config.evaluateConditions(indexCreationDate, numDocs, indexSize)) {
+            executeRollover(alias)
+        } else {
+            info = mapOf("message" to "Attempting to rollover")
+        }
+    }
+
+    private suspend fun executeRollover(alias: String) {
+        try {
+            val request = RolloverRequest(alias, null)
+            val response: RolloverResponse = client.admin().indices().suspendUntil { rolloverIndex(request, it) }
+
+            // Do not need to check for isRolledOver as we are not passing any conditions or dryrun=true
+            // which are the only two ways it comes back as false
+
+            // If response isAcknowledged it means the index was created and alias was added to new index
+            if (response.isAcknowledged) {
+                stepCompleted = true
+                info = mapOf("message" to "Rolled over index")
+            } else {
+                // If the alias update response is NOT acknowledged we will get back isAcknowledged=false
+                // This means the new index was created but we failed to swap the alias
+                failed = true
+                info = mapOf("message" to "New index created (${response.newIndex}), but failed to update alias")
+            }
+        } catch (error: Exception) {
+            failed = true
+            val mutableInfo = mutableMapOf("message" to "Failed to rollover index")
+            val errorMessage = error.message
+            if (errorMessage != null) mutableInfo.put("cause", errorMessage)
+            info = mutableInfo.toMap()
+        }
+    }
+
+    private fun getAlias(): String? {
+        val alias = clusterService.state().metaData().index(managedIndexMetaData.index).getRolloverAlias()
+
+        if (alias == null) {
+            failed = true
+            info = mapOf("message" to "There is no valid rollover_alias=$alias set on ${managedIndexMetaData.index}")
+        }
+
+        return alias
+    }
+
+    private suspend fun getIndexStats(): IndicesStatsResponse? {
+        try {
+            val statsRequest = IndicesStatsRequest()
+                    .indices(managedIndexMetaData.index).clear().docs(true)
+            val statsResponse: IndicesStatsResponse = client.admin().indices().suspendUntil { stats(statsRequest, it) }
+
+            if (statsResponse.status == RestStatus.OK) {
+                return statsResponse
+            }
+
+            failed = true
+            info = mapOf(
```

Comment:
Similar comment to `getAlias` here. I think function name shows that only thing it should do it `getIndexStats` but really it does a bit more than that which and disrupt the flow of the `execute` function. In the `execute`

```
        val statsResponse = getIndexStats()

        statsResponse ?: return
```

seems like we are not handling the null case gracefully only to find out there is some error handling logic in `getIndexStats`. Lets make `get` functions only do `get` and let the error handling logic be in the flow of the `execute`.